src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -0,0 +1,157 @@
+        }
+    }
+
+    private fun processWiderLanguagePromotion(source: FileEditorManager, file: VirtualFile) {
+        val notifications: SonarLintProjectNotifications = getSonarLintProjectNotifications(source)
+
+        val extension = file.extension
+
+        for (language in promotedLanguages.iterator()) {
+            for (suffixExtension in language.defaultFileSuffixes) {
+                if (suffixExtension.startsWith(".")) {
+                    if (suffixExtension == ".$extension") {
+                        showWiderLanguagePromotion(notifications, language.label)
+                    }
+                } else {
+                    if (suffixExtension == extension) {
+                        showWiderLanguagePromotion(notifications, language.label)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun getSonarLintProjectNotifications(source: FileEditorManager): SonarLintProjectNotifications =
+        SonarLintUtils.getService<SonarLintProjectNotifications>(
+            source.project,
+            SonarLintProjectNotifications::class.java
+        )
+
+    private fun showWiderLanguagePromotion(notifications: SonarLintProjectNotifications, language: String) {
+        val lastPromotionDate: LocalDateTime? = getLastPromotionDate(dateTimeFormatter)
+
+        if (shouldNotify(lastPromotionDate)) {
+            notifications.notifyWiderLanguageSupport(language)
+            Settings.getGlobalSettings().lastPromotionNotificationDate = LocalDateTime.now().format(dateTimeFormatter);
+        }
+    }
+
+    private fun showAdvancedRulesPromotion(notifications: SonarLintProjectNotifications, language: String) {
+        val lastPromotionDate: LocalDateTime? = getLastPromotionDate(dateTimeFormatter)
+
+        if (shouldNotify(lastPromotionDate)) {
+            notifications.notifyAdvancedRulesSupport(language)
+            Settings.getGlobalSettings().lastPromotionNotificationDate = LocalDateTime.now().format(dateTimeFormatter);
+        }
+    }
+
+    private fun getLastPromotionDate(dateTimeFormatter: DateTimeFormatter?): LocalDateTime? {
+        val lastModifiedDateString = Settings.getGlobalSettings().lastPromotionNotificationDate
+        var lastPromotionDate: LocalDateTime? = null
+
+        if (lastModifiedDateString != null) {
+            lastPromotionDate = LocalDateTime.parse(lastModifiedDateString, dateTimeFormatter)
+
+        }
+        return lastPromotionDate
+    }
+
+    private fun shouldNotify(lastPromotionDate: LocalDateTime?): Boolean {
+        val isDontAskAgain = Settings.getGlobalSettings().isPromotionDisabled
+
+        if (isDontAskAgain) {
+            return false;
+        }
+
+        if (lastPromotionDate == null) {
+            return true
+        } else {
+            val dateDiff = ChronoUnit.SECONDS.between(lastPromotionDate, LocalDateTime.now())
+
+            if (dateDiff > PROMOTION_PERIOD) {
+                return true
+            }
+        }
```

Comment:
I think this check is wrong, we are checking SECONDS which I think is wrong. You should have a look at a previous comment I made, `Instant` has a `isBefore()` method, and the code would read better